template/addons/next-auth/middleware.ts

```diff
@@ -0,0 +1,20 @@
+// Example of a restricted route that only authenticated users can access https://next-auth.js.org/configuration/nextjs#middleware
+
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+import { getToken } from "next-auth/jwt";
+
+export async function middleware(request: NextRequest) {
+  const token = await getToken({
+    req: request,
+    secret: process.env.NEXTAUTH_SECRET,
+  });
+
+  if (!token) {
+    return NextResponse.redirect(new URL("/", request.url));
```

Comment:
Tested this locally (and on a test-repo on Vercel) and I get an error that I got redirected too many times. Have you tested this successfully? Using the [other redirect method](https://nextjs.org/docs/messages/middleware-relative-urls) seems to work better.